src/main/cpp/t04_both2.cpp

```diff
@@ -23,9 +23,35 @@
 
 #include "t04_both2.h"
 #include <iostream>
+#include <set>
+#include <algorithm>
 
 using namespace std;
 
 int t04_both2() {
+    set<int> s1,s2,s3;
+    int n,m,a;
+
+    cin >> n;
+    for(int i = 0; i < n; i++){
+        cin >> a;
+        s1.insert(a);
+    }
+    cin >> m;
+
+    for (int i = 0; i < m; i++) {
+        cin >> a;
+        if (s1.find(a) != s1.end()){
+            //cout << a << " ";
+            s3.insert(a);
+        }
+    }
+
+    set <int> :: iterator it = s3.begin();
+    while (it != s3.end()) {
+        cout << *it << " ";
+        it++;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prefer prefix ++/-- operators for non-primitive types.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4594561)